File: DemoServer/Demos/Advanced/DeletingAttachments.cs
using DemoServer.Controllers;
using DemoServer.Helpers;
using Microsoft.AspNetCore.Mvc;

namespace DemoServer.Demos.Advanced
{
    public partial class AdvancedController : BaseController
    {
        [HttpGet]
        [Route("/advanced/deletingAttachments")]
        [Demo("Deleting Attachments", DemoOutputType.String, demoOrder: 216)]
        public object DeletingAttachments(string attachmentName = "the_big_cheese_logo.png")
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                session.Advanced.Attachments.Delete("companies/77-A", attachmentName);

                session.SaveChanges();
            }
            
            return $"{attachmentName} deleted successfully";
        }
    }
}

File: DemoServer/PythonDemo/demo_methods/Advanced/DeletingAttachments.py
from demo.holder import DocumentStoreHolder


class DeletingAttachments(object):
    @staticmethod
    def deleting_attachments(attachmentName = "the_big_cheese_logo.png"):
		with DocumentStoreHolder.get_store().open_session() as session:
			session.advanced.attachment.delete("companies/77-A", name=attachmentName)
			session.save_changes()

        return "{0} deleted successfully".format(attachmentName);
